addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -109,6 +110,65 @@ def test_mailing_computed_fields_default(self):
         })
         self.assertEqual(literal_eval(mailing.mailing_domain), [('email', 'ilike', 'test.example.com')])
 
+    @users('user_marketing')
+    def test_mailing_generated_source_name(self):
+        today_str = date.today().strftime('%Y-%m-%d')
+        mailing = self.env['mailing.mailing'].create({
```

Comment:
Why not using a "create multi" syntax for the 2 first to make sure that use case works:

```py
[mailing, mailing2] = self.env['mailing.mailing'].create([{
    'subject': 'Test source',
    'mailing_type': 'mail',
    'body_html': '<p>Hello ${object.name}</p>',
    'mailing_model_id': self.env['ir.model']._get('res.partner').id,
}, {
    'subject': 'Test source',
    'mailing_type': 'mail',
    'body_html': '<p>Hello ${object.name}</p>',
     'mailing_model_id': self.env['ir.model']._get('res.partner').id
}])
```